### PR TITLE
Proposing exposing as public api the various arch dependent

### PR DESCRIPTION
**Pull request by devnexen**
pause instructions so could be used in app infinite loops.
A handful of games do already so we unify it in one place.

### CODE REVIEW

**Comment by slouken** [comment]
@icculus, this seems reasonable to me, though I'm not sure about the name. What do you think?

**Comment by flibitijibibo** [comment]
We have something like this in [BRUTE](http://brute.rocks/) so it would definitely be welcome - the NDA platforms I'm aware of fully support this too.

File: include/SDL_atomic.h [hunk 1]
```diff
@@ -237,6 +237,26 @@ typedef void (*SDL_KernelMemoryBarrierFunc)();
...
+    #define SDL_CpuPause() __asm__ __volatile__("yield" ::: "memory")
+#elif (defined(__powerpc__) || defined(__powerpc64__))
+    #define SDL_CpuPause() __asm__ __volatile__("or 27,27,27");
+#elif defined(_MSC_VER) && (defined(_M_IX86) || defined(_M_X64))
+    #define SDL_CpuPause() _mm_pause()  /* this is actually "rep nop" and not a SIMD instruction. No inline asm in MSVC x86-64! */
+#elif defined(_MSC_VER) && (defined(_M_ARM) || defined(_M_ARM64))
+    #define SDL_CpuPause() __yield()
+#elif defined(__WATCOMC__) && defined(__386__)
+    /* watcom assembler rejects PAUSE if CPU < i686, and it refuses REP NOP as an invalid combination. Hardcode the bytes.  */
+    extern __inline void PAUSE_INSTRUCTION(void);
```
Comment (sezero) [inline include/SDL_atomic.h:253]: This should read `extern __inline void SDL_CpuPause(void);`
(you forgot renaming `PAUSE_INSTRUCTION` in there.)

Comment (devnexen) [reply include/SDL_atomic.h]: good point

**Comment by icculus** [comment]
> this seems reasonable to me, though I'm not sure about the name. What do you think?

Same. SDL_CpuPauseOpcode? SDL_CpuPauseInstruction?

**Comment by slouken** [comment]
> > this seems reasonable to me, though I'm not sure about the name. What do you think?
> 
> Same. SDL_CpuPauseOpcode? SDL_CpuPauseInstruction?

I like SDL_CpuPauseInstruction... or maybe SDL_CPUPauseInstruction?

**Comment by icculus** [comment]
@devnexen Update the pull request to call it SDL_CPUPauseInstruction and we'll merge!

**Pull request merged**